### PR TITLE
Emags emag emags II: electric boogaloo

### PR DESCRIPTION
**Pull request by ghost**
Emags emag emags
Emags emag emags
Emags emag emags
Emags emag emags
Emags emag emags
Emags emag emags
Emags emag emags
Emags emag emags
Emags emag emags
Emags emag emags

:cl:  
rscadd: Emags emag emags
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
The code could definitely be optimized and I'm down for suggestions.  I'm a retard and this seems like the best way for what I was doing.

**Comment by ghost** [comment]
It has been brought to my attention people can have more than two hands.  I will return.

File: code/game/objects/items/cards_ids.dm [hunk 1]
```diff
@@ -80,6 +80,25 @@
 	righthand_file = 'icons/mob/inhands/equipment/idcards_righthand.dmi'
 	item_flags = NO_MAT_REDEMPTION | NOBLUDGEON
 	var/prox_check = TRUE //If the emag requires you to be in range
+	
+/obj/item/card/emag/emag_act(mob/user)
+	var/otherEmag
+
+	if(user.held_items[1])
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline code/game/objects/items/cards_ids.dm]: you can replace all of this with
 ```
var/otherEmag = user.get_active_held_item()
if(!otherEmag)
    return
```

File: code/game/objects/items/cards_ids.dm [hunk 2]
```diff
@@ -80,6 +80,25 @@
...
+			if(user.held_items[1] != src)
+				otherEmag = user.held_items[1]
+	if(!otherEmag)
+		if(user.held_items[2])
+			if(istype(user.held_items[2], /obj/item/card/emag))
+				if(user.held_items[2] != src)
+					otherEmag = user.held_items[2]
+	if(!otherEmag)
+		return
+	to_chat(user, "<span class='notice'>The cyptographic sequencers attempt to override each other inevitably destroying themselves.</span>")
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline code/game/objects/items/cards_ids.dm]: I'd replace inevitably with something else. Maybe "override each other before destroying"?

File: code/game/objects/items/cards_ids.dm (hunk 1 above)
Comment (ghost) [reply code/game/objects/items/cards_ids.dm]: Oh my god.... it's perfect.

File: code/game/objects/items/cards_ids.dm (hunk 2 above)
Comment (ghost) [reply code/game/objects/items/cards_ids.dm]: Everyone is an english major around here.

File: yogstation/code/game/objects/items/cards_ids.dm [hunk 3]
```diff
@@ -65,3 +65,14 @@
 
 /obj/item/card/id/captains_spare
 	icon_state = "id_gold"
+
+/obj/item/card/emag/emag_act(mob/user)
+	var/otherEmag
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline yogstation/code/game/objects/items/cards_ids.dm]: remove this please

**Comment by Nickvr628** [comment]
But why?

**Comment by ghost** [comment]
Moving stuff from pre-rebase over 🐙

**Review by ThatLing — APPROVED** [review]

**Comment by jcatjmeow** [comment]
morrow asked me to harras someone to get this out faster so uh u smell and ur face bad merge

**Comment by Amelia0010** [comment]
morrow can wait 24 hours like everyone else :)

**Comment by ghost** [comment]
morrow is an impatient boy and did not know there was a 24 hour hold

**Comment by Amelia0010** [comment]
there isn't, but if we merge anything before then people complain they don't get the chance to provide input

**Pull request merged**